packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -406,14 +402,45 @@ export async function getLinkRemovalByDidExtrinsic(
 /**
  * Return the default sign callback, which uses the address argument to crete a signing closure for the given payload.
  *
- * @param keyring The keyring to retrieve the signing key.
+ * @param keypair The keypair to sign the data with.
  * @returns The signature generating callback that uses the keyring to sign the input payload using the input address.
  */
-export function defaultSignerCallback(keyring: Keyring): LinkingSignerCallback {
-  return (payload: HexString, address: Address): Promise<HexString> =>
-    Promise.resolve(
-      u8aToHex(keyring.getPair(address).sign(payload, { withType: false }))
-    )
+export function makeLinkingSignCallback(
+  keypair: KeyringPair
+): LinkingSignCallback {
+  return async function sign(payload: HexString): Promise<Uint8Array> {
+    return keypair.sign(payload, { withType: false })
+  }
+}
+
+function getUnprefixedSignature(
+  message: HexString,
+  signature: Uint8Array,
+  address: Address
+): { signature: Uint8Array; type: KeypairType } {
+  try {
+    // try to verify the signature without the prefix first
+    const unprefixed = signature.subarray(1)
+    const { crypto, isValid } = signatureVerify(message, unprefixed, address)
+    if (isValid) {
+      return {
+        signature: unprefixed,
+        type: crypto as KeypairType,
+      }
+    }
+  } catch {
+    // if it fails, maybe the signature prefix caused that, so we try to verify the whole signature
+  }
+
+  const { crypto, isValid } = signatureVerify(message, signature, address)
+  if (isValid) {
+    return {
+      signature,
+      type: crypto as KeypairType,
+    }
+  }
+
+  throw new SDKErrors.SignatureUnverifiableError()
```

Comment:
I replaced the polkadot asserts (not used anywhere else in the SDK) with a proper SDK error

Reply:
makes sense